### PR TITLE
feat: add dependency management

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
## Description

Adds capability to add dependencies for policy assignments and policy role assignments.

This provides a nice workaround to the limitation of not being able to pass unknown values into the alz provider.

## Type of Change

- [ ] Non-module change (e.g. CI/CD, documentation, etc.)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates.
  - [ ] Breaking changes.
  - [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

**Comment by c-baumgartner** [comment]
Hi @matt-FFFFFF , 

I ❤️ this. The idea to use `terraform_data`  and `sha256` as some kind of "intermediate" to have a chain `depends_on` can rely on - awesome!

I am sure this will, together with the better retry logic of AzAPI v2, help a lot!

**Comment by matt-FFFFFF** [comment]
Thanks @c-baumgartner!

`terraform_data` might be my new favourite resource 😂

File: variables.tf [hunk 1]
```diff
@@ -33,6 +33,33 @@ DESCRIPTION
...
+  type = object({
+    policy_role_assignments = optional(any, null)
+    policy_assignments      = optional(any, null)
+  })
+  default     = {}
+  description = <<DESCRIPTION
+Place dependent values into this variable to ensure that resources are created in the correct order.
+Ensure that the values placed here are computed/known after apply, e.g. the resource ids.
+
+This is necessary as the `depends_on` attribute is not supported bu this module as we use the alz provider.
```
Comment (jtracey93) [inline variables.tf:46]:
```suggestion
This is necessary as the `depends_on` attribute is not supported but this module as we use the alz provider.
```

File: _header.md [hunk 2]
```diff
@@ -74,6 +82,12 @@ module "example" {
 }
 ```
 
+### `var.dependencies`
+
+This variable is used as a workaround for the lack of support for `depends_on` in the ALZ provider.
+Place values into this variable to ensure that policies and policy role assignmetns do nto get created until dependent resources are available.
```
Comment (jtracey93) [inline _header.md:88]:
```suggestion
Place values into this variable to ensure that policies and policy role assignments do not get created until dependent resources are available.
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]

**Review by jaredfholgate — APPROVED** [review]
LGTM

**Review by jaredfholgate — DISMISSED** [review]

**Review by jaredfholgate — APPROVED** [review]

**Review by jchancellor-ms — APPROVED** [review]
Approved

**Pull request merged**